geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSpark.scala

```diff
@@ -8,32 +8,76 @@
 
 package org.locationtech.geomesa.spark
 
+import java.io.{BufferedWriter, StringWriter}
 import java.util.ServiceLoader
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.spark.SparkContext
 import org.apache.spark.rdd.RDD
 import org.geotools.data.Query
-import org.opengis.feature.simple.SimpleFeature
+import org.geotools.geojson.feature.FeatureJSON
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+import scala.reflect._
 
 trait SpatialRDDProvider {
+  import TypeDefault._
+
+  type GeoJSONString = String
+
+  def toScalaMap(in: SpatialRDD): RDD[Map[String, AnyRef]] = {
+    val keys = in.schema.getAttributeDescriptors.map(_.getName).map(_.getLocalPart)
+    in.rdd.map(sf => keys.zip(sf.getAttributes).toMap).map(m => m + ("geom" -> m("geom").toString))
+  }
+
+  def toJavaMap(in: SpatialRDD): RDD[java.util.Map[String, AnyRef]] = {
+    toScalaMap(in).map(mapAsJavaMap)
+  }
+
+  def toGeoJSONString(in: SpatialRDD): RDD[GeoJSONString] = {
+    in.rdd.mapPartitions(features => {
+      val json = new FeatureJSON
+      val sw = new StringWriter
+      val bw = new BufferedWriter(sw)
+      features.map(f => try { json.writeFeature(f, bw); sw.toString } finally { sw.getBuffer.setLength(0) })
+    })
+  }
+
+  def transformFunc[A : ClassTag]: (SpatialRDD => RDD[A]) = (classTag[A] match {
+    case ct if ct == classTag[GeoJSONString] => toGeoJSONString _
+    case ct if ct == classTag[Map[String, AnyRef]] => toScalaMap _
+    case ct if ct == classTag[java.util.Map[String, AnyRef]] => toJavaMap _
+    case ct if ct == classTag[SimpleFeature] => in: SpatialRDD => in.rdd
+    case _ => new IllegalArgumentException
+  }).asInstanceOf[SpatialRDD => RDD[A]]
 
   def canProcess(params: java.util.Map[String, java.io.Serializable]): Boolean
 
-  def rdd(conf: Configuration,
-          sc: SparkContext,
-          dsParams: Map[String, String],
-          query: Query): RDD[SimpleFeature]
+  def rdd[T : ClassTag](conf: Configuration,
+             sc: SparkContext,
+             params: Map[String, String],
+             query: Query)(implicit default: T := SimpleFeature): RDD[T] = {
+      val tf = transformFunc[T]
```

Comment:
would it be worthwhile to override this in the accumulo class to use the various input formats directly and save the map step?